src/test/ui/associated-types/associated-types-overridden-binding-2.rs

```diff
@@ -1,8 +1,8 @@
+// check-pass
 #![feature(trait_alias)]
 
 trait I32Iterator = Iterator<Item = i32>;
 
 fn main() {
     let _: &dyn I32Iterator<Item = u32> = &vec![42].into_iter();
-    //~^ ERROR type mismatch
```

Comment:
I think this test changes because `ExistentialPredicate::stable_cmp` doesn't actually look at the its substs which seems kind of incorrect to me :thinking: